src/aoe_ctc.jl

```diff
@@ -0,0 +1,147 @@
+"""
+
+function ctc_aoe(aoe_all::Vector{<:Real}, ecal_all::Vector{<:Unitful.RealOrRealQuantity}, qdrift_e_all::Vector{<:Real}, compton_bands::Vector{<:Unitful.RealOrRealQuantity}, peak::Real = 0.0, window::Tuple{<:Real, <:Real} = (50.0, 8.0), hist_start::Real = -20.0, hist_end::Real = 5.0, bin_width::Real = 0.05; aoe_expression::Union{Symbol, String}="aoe", e_expression::Union{Symbol, String} = "e", pseudo_prior::NamedTupleDist = NamedTupleDist(empty = true), pseudo_prior_all::NamedTupleDist = NamedTupleDist(empty = true), pol_order::Int=1) # deleted m_cal since no calibration
+
+    # distribution compton band entries
+    compton_start = ustrip.(compton_bands)
+    compton_end = compton_start .+ 20
+    # create mask (make sure everything is unitless)
+    mask = falses(length(ustrip.(ecal_all)))
+    for i in 1:length(compton_start)
+        mask .|= compton_start[i] .<= ustrip.(ecal_all) .<= compton_end[i]
+    end
+    # Filter the energies, aoe and qdrift values based on the mask
+    aoe = aoe_all[mask]
+    qdrift_e = qdrift_e_all[mask]
+
+    # create cut window around peak
+    cut = peak - first(window) .< aoe .< peak + last(window)
+    @debug "Cut window: $(peak - first(window)) < aoe < $(peak + last(window))"
+    aoe_cut, qdrift_e_cut = aoe[cut], qdrift_e[cut]
+    
+    # calculate optimal bin width (if needed for other purposes)
+    bin_width_window = 5.0 ### this parameter might be modified since it's copied from the energy case
+    bin_width        = get_friedman_diaconis_bin_width(aoe[peak - bin_width_window .< aoe .< peak + bin_width_window]) ### or use 0.05
+    bin_width_qdrift = get_friedman_diaconis_bin_width(qdrift_e[peak - bin_width_window .< aoe .< peak + bin_width_window])
+
+
+    # get σ before correction
+    # fit peak (with the aoe_two_bck fit function which uses two EMGs)
+    h_before = fit(Histogram, aoe_cut, hist_start:bin_width:hist_end) ### the current values: hist_start -20 and hist_end 3
+    ps_before = estimate_single_peak_stats(h_before)
+    result_before, report_before = fit_single_aoe_compton(h_before, ps_before, fit_func=:aoe_two_bck, pseudo_prior=pseudo_prior, uncertainty=true)
+    @debug "Found Best σ before correction: $(round(result_before.σ, digits=2))"
+
+    # create optimization function
+    function f_optimize_ctc(fct, aoe, qdrift_e, bin_width)
+        # calculate drift time corrected aoe
+        aoe_ctc = aoe .+ PolCalFunc(0.0, fct...).(qdrift_e)
+        # fit peak
+        h = fit(Histogram, aoe_ctc, minimum(aoe_ctc):bin_width:maximum(aoe_ctc))
+        ps = estimate_single_peak_stats(h)
+        result_peak, report_peak = fit_single_aoe_compton(h, ps, fit_func=:aoe_two_bck, pseudo_prior=pseudo_prior, uncertainty=false) ### in ctc.jl uncertainty false is used (but I don't know why)
+        # get σ and peak height
+        μ = mvalue(result_peak.μ)
+        σ = mvalue(result_peak.σ)
+        p_height = maximum(report_peak.f_fit.(μ-0.2*σ:0.001:μ+0.2*σ)) # hardcoded 0.001 for now
```

Comment:
hard-coded `0.001` to determine the maximum. Assuming a normalization to `μ = 0 `and `σ = 1`, this should be sufficiently fine.